### PR TITLE
feat: Regenerate k8s

### PR DESCRIPTION
**Pull request by yevgenypats**
Given multiple requests from number of key users around incompleteness  of k8s I updated our codegen to do both fetcher and mocks. k8s API is especially good and consistent so it was relatively straight forward to have 99% auto-generated source plugin apart of a few places with custom resolver or overrides (that are still done on the codegen level).

This doesn't have any breaking changes but things to consider in the future is how to handle v1,v1beta,v2 apis. We might want to introduce those into our tables as well but can be done in a future PR.

Once this is merged it will also help with the WIP policy for k8s.

### CODE REVIEW

File: plugins/source/k8s/client/client.go [hunk 1]
```diff
@@ -52,14 +52,14 @@ func (c Client) WithContext(k8sContext string) *Client {
...
-	c.contexts = contexts
-}
+// func (c *Client) SetServices(s map[string]Services) {
+// 	c.services = s
+// 	contexts := make([]string, 0, len(s))
+// 	for k := range s {
+// 		contexts = append(contexts, k)
+// 	}
+// 	c.contexts = contexts
+// }
```
Comment (erezrokah) [inline plugins/source/k8s/client/client.go:62]:
```suggestion
```

Can we delete this if not used?

File: plugins/source/k8s/gen.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+mockgen --package=mocks --destination="client/mocks/mock.go" "k8s.io/client-go/kubernetes" Interface
```
Comment (erezrokah) [inline plugins/source/k8s/gen.sh:1]: Maybe we can put this in the Makefile under:
https://github.com/cloudquery/cloudquery/blob/43d5fcd77160c4803f88ddce55cd96969da66e21/plugins/source/k8s/Makefile#L25

or `gen-mocks:` and update `gen:`?

**Review by erezrokah — APPROVED** [review]
Nice 🚀  Added 2 non blocking comments

File: plugins/source/k8s/Makefile [hunk 3]
```diff
@@ -1,7 +1,7 @@
 # Generate mocks for mock/unit testing 
 .PHONY: generate-mocks
-generate-mocks:
-	go generate ./client/services/...
+gen-mocks:
```
Comment (erezrokah) [inline plugins/source/k8s/Makefile:3]: Can we add this to the `gen` target?

File: plugins/source/k8s/codegen/recipes/templates/mock.go.tpl [hunk 4]
```diff
@@ -0,0 +1,51 @@
+// Code generated by codegen; DO NOT EDIT.
```
Comment (shimonp21) [inline plugins/source/k8s/codegen/recipes/templates/mock.go.tpl:1]: nit: rename file to `mock_test.go.tpl`

File: plugins/source/k8s/Makefile (hunk 3 above)
Comment (yevgenypats) [reply plugins/source/k8s/Makefile:3]: Added

Comment (erezrokah) [reply plugins/source/k8s/Makefile:3]: Where :) ? I don't see a new commit

**Comment by erezrokah** [comment]
@yevgenypats I added https://github.com/cloudquery/cloudquery/pull/4495/commits/09bbd5d5515e708fa790efcb1a642a47ac2fac7a per my comment. I hope that makes sense

**Pull request merged**